src/models/CommonModel.ts

```diff
@@ -106,12 +202,14 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * @param mergeFrom CommonModel to merge from
    * @param originalSchema 
    */
-  private static mergeItems(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema) {
+  private static mergeItems(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema, alreadyIteratedModels: Map<CommonModel, CommonModel> = new Map()) {
     const merge = (models: CommonModel | CommonModel[] | undefined): CommonModel | undefined => {
       if (Array.isArray(models)) {
         if (models.length > 0) {
-          let mergedItemsModel: CommonModel = models[0];
-          models.forEach((model) => { mergedItemsModel = CommonModel.mergeCommonModels(mergedItemsModel, model, originalSchema); });
+          let mergedItemsModel: CommonModel | undefined = undefined;
+          models.forEach((model) => { 
+            mergedItemsModel = CommonModel.mergeCommonModels(mergedItemsModel, model, originalSchema, alreadyIteratedModels); 
```

Comment:
Why do not you just use the last element of an array? It is unclear why this iteration take place.